libcob/move.c

```diff
@@ -471,106 +471,199 @@ cob_move_alphanum_to_alphanum (cob_field *f1, cob_field *f2)
+	/*  complicated in that each time thru the packing loop we  */
+	/*  pack TWO DIGITS. So we divide the offset by 2 to        */
+	/*  adjust the position of the pointer to the packed field  */
+	/*  to the correct byte where the digit will be placed.     */
+	/*  Also we need to determine if the first digit to be      */
+	/*  packed will go into the HIGH ORDER nibble or the LOW    */
+	/*  ORDER nibble. If the difference in the offsets is ODD,  */
+	/*  then the first digit will need to go into the LOW       */
+	/*  ORDER nibble, so we set a switch for that.              */
+	/*                                                          */
+	/*  Before starting the packing loop we need to check if    */
+	/*  the first digit has to go into the LOW ORDER nibble,    */
+	/*  if so we need to do that first.                         */
+	/*                                                          */
+	/*  Also after completing the packing loop, we need to      */
+	/*  check if there is a digit left to be packed, if so we   */
+	/*  need to put it into the HIGH ORDER nibble of the next   */
+	/*  location in the receiving field.                        */
+	/*                                                          */
+	/*  Then we need to clear the PAD nibble if present and     */
+	/*  set the sign nibble if the receiving field is signed.   */
+	/*                                                          */
+	/************************************************************/
+
+	const int     sign = COB_GET_SIGN_ADJUST (f1);
+	const short   scale1 = COB_FIELD_SCALE (f1);
+	const short   scale2 = COB_FIELD_SCALE (f2);
+	const int     target_no_sign_nibble = COB_FIELD_NO_SIGN_NIBBLE (f2);
+	unsigned int      f1_digits, f2_digits;
+	unsigned int      start_in_low_nibble = 0;
+	int               f1_offset, f2_offset;
+
+	register unsigned char  *p, *p_end, *q, *q_end;
+
+
+			/************************************************************/
+			/*                                                          */
+			/*  Note that when calculating the offsets of the first     */
+			/*  position in each field we need to use the number of     */
+			/*  digits in the actual data field not the number of       */
+			/*  digits returned by the COB_FIELD_DIGITS function.       */
+			/*                                                          */
+			/*  Note that this logic works whether or not there is a    */
+			/*  "P" any where in the PICTURE clause.                    */
+			/*                                                          */
+			/************************************************************/
+
+	if (COB_FIELD_SIGN_SEPARATE (f1))
+			f1_digits = f1->size - 1;
+	else
+			f1_digits = f1->size;
+
+	p = COB_FIELD_DATA (f1);
+	p_end = p + f1_digits - 1;
+
+	f1_offset = f1_digits - scale1 - 1;
+
+
+	if (target_no_sign_nibble)
+			f2_digits = f2->size << 1;
+	else
+			f2_digits = (f2->size << 1) - 1;
+
+	q = f2->data;
+	q_end = q + f2->size - 1;
+
```

Comment:
If you can spot (git/svn blame) the commit that reformated this in 3.x (and _possibly_ includes a fix as well), then please get it into the bunch if easily possible - if not, then it gets in later.

Reply:
There is no commit in 3.x that reformats this yet.
I do intend to fix that in 3.x and merge that in 4.x (probably when the ~50 remaining commits are merged, since there are a few more of these commits with wrong formatting to come).